code/workspaces/infrastructure-api/src/controllers/nameController.js

```diff
@@ -3,10 +3,10 @@ import stackRepository from '../dataaccess/stacksRepository';
 import dataStorageRepository from '../dataaccess/dataStorageRepository';
 
 async function isUnique(request, response) {
-  const { name } = matchedData(request);
+  const { name, projectKey } = matchedData(request);
 
-  const stack = await stackRepository.getOneByName(request.user, name);
-  const volume = await dataStorageRepository.getAllByName(request.user, name);
+  const stack = await stackRepository.getOneByName(projectKey, request.user, name);
+  const volume = await dataStorageRepository.getAllByName(request.user, name); // needs to be updated
```

Comment:
Does this mean it needs to be updated in another PR to tie in with Andy's stuff?

Reply:
Yeah, not sure what the exact function signiture that Andy has for this stage. Will merge in master and take a look otherwise it will need to be updated once this function is updated.